modules.d/90dmsquash-live-autooverlay/create-overlay.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+
+type getarg > /dev/null 2>&1 || . /lib/dracut-lib.sh
+
+if getargbool 0 rd.live.debug -n -y rdlivedebug; then
+    exec > /tmp/create-overlay.$$.out
+    exec 2>> /tmp/create-overlay.$$.out
+    set -x
+fi
+
+gatherData() {
+    overlay=$(getarg rd.live.overlay)
+    if [ -z "$overlay" ]; then
+        info "Skipping overlay creation: kernel command line parameter 'rd.live.overlay' is not set"
+        exit 0
+    fi
+    # shellcheck disable=SC2086
+    if ! str_starts ${overlay} LABEL=; then
+        die "Overlay creation failed: the partition must be set by LABEL in the 'rd.live.overlay' kernel parameter"
```

Comment:
Sorry for not catching this earlier, but have you had a chance to test this with 
rd.live.overlay=none
rd.live.overlay=/dev/sdb1:persistent-overlay.img
rd.live.overlay=UUID=99440c1f-8daa-41bf-b965-b7240a8996f4

As an example according to the man page
If rd.live.overlay=none is provided as a kernel command
line option, a writable, linear Device-mapper target is
created on boot with no overlay.

It seems just packaging this new module would auto-install itself and this line would be executed (and fail with an error massage) at each boot (except when rd.live.overlay=LABEL" is used. 

I am not suggesting that this additional code would break the functionality for "rd.live.overlay=none", but perhaps there is a more inclusive way to test if overlay creation is expected. Also perhaps this should be an info and not a die level message ?

Reply:
> Sorry for not catching this earlier, but have you had a chance to test this with
> rd.live.overlay=none
> rd.live.overlay=/dev/sdb1
> rd.live.overlay=UUID=99440c1f-8daa-41bf-b965-b7240a8996f4

It doesn't make sense to use this module with those settings:
* `none` would disable the use of overlay, making it pointless to include this module
* specifying an existing partition doesn't make sense because this module is meant to automatically create the partition in the free space on the boot medium
* specifying the UUID doesn't make sense because it will be created when this module formats the new partition's filesystem

> It seems just packaging this new module would auto-install itself

I was under the impression that it would only be included in the initramfs if the module was specifically listed in the configuration files or on the `dracut` command line. Did I misinterpret something?

The intent is for this module to only be included for certain builds (my use case is live ISO images booted on USB sticks) by placing a drop-in configuration file in `/etc/dracut.conf.d/` like so:
https://github.com/iammattcoleman/overlay-test/blob/fedora-36/image-description/root/etc/dracut.conf.d/dmsquash-live-autooverlay.conf

> Also perhaps this should be an info and not a die level message ?

It's a `die` level message because I figured that including this module and not properly specifying the `rd.live.overlay` value would only happen by mistake. I can't think of a scenario where someone would include this module in an initramfs but not want it to work.

Reply:
> I was under the impression that it would only be included in the initramfs if the module was specifically listed in the configuration files or on the `dracut` command line.

Sorry, I missed that you're returning 255 in https://github.com/dracutdevs/dracut/pull/1991/files#diff-19cd0a77b761132c9ca0494191b45ce737857a6fc21dbed98650b528ada2b60em which will do exactly what you stated.